src/layers.rs

```diff
@@ -141,3 +141,56 @@ impl Module for LayerNorm {
         )
     }
 }
+
+/// Linear fully-connected layer.
+#[derive(Debug)]
+pub struct Linear {
```

Comment:
Why implement our own `Linear`? Does `tch::nn::Linear` not offer some things we need?

Reply:
It does, but it has two downsides:

* We can't directly implement additional methods on it (`from_tensor` in this case), unless we define them as additional traits.
* It expects an `[out_dim, in_dim]` weight matrix and then transposes them during actual matrix multiplication. The BERT matrices are `[in_dim, out_dim]`. Seemed like a bit of a waste to transpose them when initializing the `Linear` layer and then transposing them for each `forward`. Though I guess in the implementation the double transpose results in a normal tensor again, since tansposes are probably a tensor attribute and not an actual transpose.

Reply:
Relevant issue: https://github.com/pytorch/pytorch/issues/2159

So, `tch-rs` just seems to follow the layout of PyTorch and PyTorch uses a transposed matrix for historical reasons.

Reply:
I see, wondering whether some of this could go into `tch::nn::Linear`. I guess the transposing is something they won't agree to, PyTorch seems to be doing the same thing. OTOH, not doing the transpose may seem unexpected for someone coming from other torch projects.

`from_tensor` may be nice to have in `tch-rs`, although defining a trait may be difficult since different layers may have a varying number of weights.

Reply:
I'll rewrite it to use `Linear`, I think we can replace `from_tensor` by a single trait `PlaceInVarStore`.

Reply:
`from_tensor` is now replaced by a single `PlaceInVarStore` trait. I've worked around the variable number of parameter tensors by implementing this on the layers rather than the individual weights.

I guess for the future it would be nice to check if we can cheaply find out if a tensor is already in a `VarStore` and making this a no-op when it is. (